dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,514 @@
+support. seL4 has basic support for an old version of musl C library (v1.1.16).
+
+Using seL4 would require a significant amount of work:
+- Extending USB drivers
+- Implementing USB EEM driver
+- Implementing TPM driver
+- Porting [fobnail-attester](https://github.com/fobnail/fobnail-attester)
+- seL4 runs all processes in unprivileged mode, so the kernel itself would have
+  to be modified to allow booting of another kernel
+
+[Genode](https://github.com/genodelabs/genode) is an framework for creating
+custom, specialized OSes. It provides USB drivers (including `usb-net` driver
+from which should handle USB EEM) and a network stack. Its support for seL4 used
+to be incomplete, and many components were broken. However, that might have
+changed, and Genode may be an easier way to get software running on seL4. We
+have opened an [issue](https://github.com/genodelabs/genode/issues/4480) here
+too.
+
+### Linux
+
+In the case of Linux, a minimal distribution will be prepared that meets the
+requirements of the project. [Yocto Project](https://www.yoctoproject.org/) will
+be used for this, because it gives a lot of freedom in manipulating the elements
+that make up the target system.
+
+#### Running in DLME
+
+Ensuring that the operating system works in DLME can be achieved by using the
+[TrenchBoot](https://trenchboot.org/) project. For this purpose, a Yocto
+[meta-fobnail](https://github.com/fobnail/meta-fobnail) layer has been created
+that integrates the necessary elements. The effects of the tests are presented
+in a [separate document](./running-os-in-dlme.md).
+
+#### Fobnail integration
+
+At this point, the [fobnail-attester](https://github.com/fobnail/fobnail-attester)
+application is strongly dependent on Linux. Mainly due to the fact that its
+development took place on this operating system. Therefore, it is important that
+the created minimal OS also has an integrated attester application. The main
+dependencies of its operation are drivers for USB, USB EEM and TPM, but running
+them under Linux will not cause major problems.
+
+### Little Kernel / Trusted Little Kernel
+
+Little Kernel is a small, embedded monolithic kernel that runs on x86, ARM and
+RISC-V among others, provides a custom very basic libc. LK focuses on ARM
+platforms and lacks drivers for devices typically found in x86 platforms. LK has
+been used as a base for aboot (Android bootloader) and
+[Fuchsia OS](https://fuchsia.dev/).
+
+Trusted Little Kernel (available at
+`https://nv-tegra.nvidia.com/r/3rdparty/ote_partner/tlk.git`, not accessible
+through web browser - must be cloned with `git`) is a fork of LK intended to be
+run as TEE OS. It is developed by NVIDIA for Tegra platforms, support for all
+other platforms has been removed.
+
+#### Running in DLME
+
+LK boots over Multiboot 1 which is not supported by SKL. The problem is similar
+to that with Zephyr (see [Zephyr](#running-in-dlme) section for detail). However
+Multiboot2 support could be easily added.
```

Comment:
```suggestion
Multiboot2 support could be added.
```

I would avoid judgment at this point, since from my experience those claims are typically not true, but drive price down.

Reply:
Removed